### PR TITLE
bsunanda:Run2-alca54 Backport HcalIsolatedBunch from 8_1_X

### PR DESCRIPTION
**Pull request by bsunanda**
Add HcalIsolatedBunchFilter and HcalIsolatedBunchSelector + associated DQM code for proposed HcalIsolatedBunch trigger path

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaIsolatedBunchFilter.cc [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+#endif
+  //Step1: Find if the event passes one of the chosen triggers
+  if ((trigIsoBunchNames_.size() == 0) && (trigJetNames_.size() == 0)) {
+    accept = true;
+  } else {
+    /////////////////////////////TriggerResults
+    edm::Handle<edm::TriggerResults> triggerResults;
+    iEvent.getByToken(tok_trigRes_, triggerResults);
+    if (triggerResults.isValid()) {
+      std::vector<std::string> modules;
```
Comment (mmusich) [inline Calibration/HcalAlCaRecoProducers/plugins/AlCaIsolatedBunchFilter.cc]: @bsunanda this is not used

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaIsolatedBunchSelector.cc [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  edm::LogInfo("AlCaIsoBunch") << "Run " << iEvent.id().run() << " Event " 
+			       << iEvent.id().event() << " Luminosity " 
+			       << iEvent.luminosityBlock() << " Bunch " 
+			       << iEvent.bunchCrossing() << std::endl;
+#endif
+  //Step1: Find if the event passes the chosen trigger
+  edm::Handle<edm::TriggerResults> triggerResults;
+  iEvent.getByToken(tok_trigRes_, triggerResults);
+  if (triggerResults.isValid()) {
+    std::vector<std::string> modules;
```
Comment (mmusich) [inline Calibration/HcalAlCaRecoProducers/plugins/AlCaIsolatedBunchSelector.cc]: here as well `modules` is not used

File: DQMOffline/CalibCalo/src/DQMHcalIsolatedBunchAlCaReco.cc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+void DQMHcalIsolatedBunchAlCaReco::analyze(const edm::Event& iEvent, 
+					   const edm::EventSetup& iSetup ){  
+ 
+  bool accept(false);
+  /////////////////////////////TriggerResults
+  edm::Handle<edm::TriggerResults> triggerResults;
+  iEvent.getByToken(trigResult_, triggerResults);
+  if (triggerResults.isValid()) {
+    std::vector<std::string> modules;
```
Comment (mmusich) [inline DQMOffline/CalibCalo/src/DQMHcalIsolatedBunchAlCaReco.cc]: and finally here

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Configuration/AlCa/python/autoAlca.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 AlCaRecoMatrix = {'AlCaLumiPixels' : 'LumiPixels',
                   'Charmonium'     : 'TkAlJpsiMuMu',
-                  'Commissioning'  : 'HcalCalIsoTrk',
+                  'Commissioning'  : 'HcalCalIsoTrk+HcalCalIsolatedBunchSelector',
                   'Cosmics'        : 'TkAlCosmics0T+MuAlGlobalCosmics+HcalCalHOCosmics+DtCalibCosmics',
```
Comment (mmusich) [inline Configuration/AlCa/python/autoAlca.py]: @bsunanda we miss the `HcalCalIsolatedBunchFilter` on the `JetHT` PD as in [81x version of autoAlCa](https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Configuration/AlCa/python/autoAlca.py)

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@mmusich Thanks for pointing this out. Corrected and put for tests

**Comment by mmusich** [comment]
@bsunanda you have conflicts in this branch now:

```
cmsrel CMSSW_8_0_X_2016-07-27-2300
cd CMSSW_8_0_X_2016-07-27-2300/src/
cmsenv
git cms-merge-topic 15292
```
yields:

```
Auto-merging Configuration/AlCa/python/autoAlca.py
CONFLICT (content): Merge conflict in Configuration/AlCa/python/autoAlca.py
Automatic merge failed; fix conflicts and then commit the result.
Unable to merge branch refs/pull/15292/head from repository cms-sw.
```

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by mmusich** [comment]
+1

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by mmusich** [comment]
+1

**Pull request merged**